### DISCUSSION TITLE
Is mkdocs-redirects supposed to work with the projects plugin?

### DISCUSSION THREAD

**Discussion by bs-juston** [squidfunk/mkdocs-material#7292] [Q&A] [answered]
We are planning to move from a single-project documentation to a more complex, multi-project approach using the (insiders) projects plugin. Now redirecting (from the old structure to the new one) using the [redirects plugin](https://github.com/mkdocs/mkdocs-redirects) does not work -- neither within a (sub) project nor beyond projects.
1. Is that supposed to work at all? (The projects plugin's [limitations](https://squidfunk.github.io/mkdocs-material/plugins/projects/#limitations) do not mention redirects ...)
2. If so, is there any configuration required in addition to the redirect map?

Thanks a lot.

**Comment by squidfunk** [comment]
Could you provide a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) that isolates and showcases the problem you're facing, so we know what you're trying to do? If we have something we can run, we can decide whether it's a bug or feature request.

**Comment by kamilkrzyskow** [comment, marked as answer]
If you want to point redirects to files outside the docs directory it will not do it as it creates an internal list of valid files within to redirect to.

However, using redirects via external links should work, just provide the proper https:// link.

**Comment by bs-juston** [comment]
Thanks for your answers.

I have created a minimum reporduction for you to reproduce the behavior. Playing around with that and in accordance with Kamil's reply, I found using external links does the trick -- so there is no immediate need to tackle this.

So, here is the feature request: It would be nice if the redirect map eventualy worked with a path(-like) pattern, the same way as the navigation section does (which shows an error on build, but the links finally work ...)

[9.5.27+insiders.4.53.11-redirects-with-projects-plugin.zip](https://github.com/user-attachments/files/15970250/9.5.27%2Binsiders.4.53.11-redirects-with-projects-plugin.zip)

Thanks a lot.

**Reply by squidfunk** [reply]
Thanks! The [redirect plugin](https://github.com/mkdocs/mkdocs-redirects) is a third-party plugin which we do not immediately control. However, yes, interlinking between projects is definitely an open problem that is difficult to solve with separate MkDocs projects. We take this example back to the drawing board.

**Reply by bs-juston** [reply]
OK.

As to "interlinking between projects": I don't know if we are talking about the same here, but that works, doesn't it? It took a while to understand how to specify the target, but, yes, you can link to files in other project trees. In the example I attached previously, you can, for example, link from concepts.md to user_fr.md using `[Documentation opérationnelle](../../user/fr/user_fr/)` -- even with anchors.

**Reply by squidfunk** [reply]
Yes, but that means you're tying one project explicitly to another project, and it's not clear whether this is the best approach. We are working towards something that allows to:

1. Interlink projects that are in the same repository and built together
2. Interlink between projects that are also built on their own. Those links would be dead in this case, so there needs to be some way to specify where all links for some project would need to go. I'd call this something like re-binding. This might be the case if you have a website with different products that you build together, but you also ship each of the product documentation on their own. If you link from one product to another, the linked would be dead. Thus, we need to find a way to specify some kind of alternate target or resolution mechanism in case a project links to another project that is out of the immediate build scope.

In the meantime, you should be good to go with the solution that works for you ☺️

**Reply by bs-juston** [reply]
Ah, I see. Well, yes, this seems to be a bit of a challenge. We never thought about that because we would always build everything from scratch, wipe the web server clean and upload all again.

Good luck, anyway!
